app/src/main/kotlin/github/com/raccok/dota2androidapp/Backend.kt

```diff
@@ -0,0 +1,138 @@
+package github.com.raccok.dota2androidapp
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.net.ConnectivityManager
+import android.util.Log
+import android.widget.Toast
+import com.loopj.android.http.AsyncHttpClient
+import com.loopj.android.http.JsonHttpResponseHandler
+import cz.msebera.android.httpclient.Header
+import org.json.JSONObject
+
+class Backend {
+  private lateinit var mFrontend: MainActivity
+  private lateinit var mSharedPreferences: SharedPreferences
+  private var mHeroNames: MutableList<String> = mutableListOf()
+
+  fun init(frontend: MainActivity) : Boolean {
```

Comment:
You generally don't want to pass `Activity` around as variables, it can lead to activity leaks and generally not a good practice. Try passing the `Context` instead, it should cover all your need. If you ever need to pass an `Activity`, then there's a high chance that you can refactor to prevent that from happening.

Reply:
Thank you very much. I am now much happier with the code after refactoring this, please check if you see another bad practice